packages/babel-parser/src/tokenizer/types.js

```diff
@@ -158,6 +158,8 @@ export const tt: { [name: string]: TokenType } = {
   ellipsis: createToken("...", { beforeExpr }),
   backQuote: createToken("`", { startsExpr }),
   dollarBraceL: createToken("${", { beforeExpr, startsExpr }),
+  templateTail: createToken("...`", { startsExpr }),
+  templateMiddle: createToken("...${", { beforeExpr, startsExpr }),
```

Comment:
While reviewing, I found it quite confusing that `templateMiddle` can also be at the start of a template (I expected `templateStart` `ŧemplateMiddle`+ `templateTail`). I'd prefer if it was just called `templateNonTail`.